indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java

```diff
@@ -0,0 +1,1021 @@
+      final File tmpDir = Paths.get(tmpDirLoc).toFile();
+
+      final DataSegment segment = ((DataSegmentSplit) split).getDataSegment();
+
+      final HadoopDruidConverterConfig config = converterConfigFromConfiguration(context.getConfiguration());
+
+      context.setStatus("DOWNLOADING");
+      context.progress();
+      final File inDir = new File(tmpDir, "in");
+      if (inDir.exists() && !inDir.delete()) {
+        log.warn("Could not delete [%s]", inDir);
+      }
+      if (!inDir.mkdir()) {
+        log.warn("Unable to make directory");
+      }
+      final File tmpDownloadFile = File.createTempFile("dataSegment", ".zip");
+      if (tmpDownloadFile.exists() && !tmpDownloadFile.delete()) {
+        log.warn("Couldn't clear out temporary file [%s]", tmpDownloadFile);
+      }
+      try {
+        final Path inPath = new Path(getURIFromSegment(segment));
+        final FileSystem fs = inPath.getFileSystem(context.getConfiguration());
+        if (fs instanceof LocalFileSystem) {
+          java.nio.file.Files.copy(new File(inPath.toUri()).toPath(), tmpDownloadFile.toPath());
+        } else {
+          fs.copyToLocalFile(false, inPath, new Path(tmpDownloadFile.toURI()));
+        }
+        // This is to bypass Guava dependencies in CompressionUtils.unzip
+        long size = 0L;
+        try (final ZipInputStream zipInputStream = new ZipInputStream(
+            new BufferedInputStream(
+                new FileInputStream(
+                    tmpDownloadFile
+                )
+            )
+        )) {
+          final byte[] buffer = new byte[1 << 13];
+          for (ZipEntry entry = zipInputStream.getNextEntry(); entry != null; entry = zipInputStream.getNextEntry()) {
+            final String fileName = entry.getName();
+            try (final FileOutputStream fos = new FileOutputStream(
+                inDir.getAbsolutePath()
+                + File.separator
+                + fileName
+            )) {
+              for (int len = zipInputStream.read(buffer); len >= 0; len = zipInputStream.read(buffer)) {
+                size += len;
+                fos.write(buffer, 0, len);
+              }
+            }
+          }
+        }
+        log.debug("Loaded %d bytes into [%s] for converting", size, inDir.getAbsolutePath());
+        context.getCounter(COUNTER_GROUP, COUNTER_LOADED)
+               .increment(size);
+      }
+      finally {
+        if (tmpDownloadFile.exists() && !tmpDownloadFile.delete()) {
+          log.warn("Temporary download file could not be deleted [%s]", tmpDownloadFile);
+        }
+      }
+      context.setStatus("CONVERTING");
+      context.progress();
+      final File outDir = new File(tmpDir, "out");
+      if (!outDir.mkdir() && (!outDir.exists() || !outDir.isDirectory())) {
+        throw new IOException(String.format("Could not create output directory [%s]", outDir));
+      }
+      IndexMaker.convert(
```

Comment:
we should look into removing IndexMaker. It was created to build v9 segments directly, but given the fact that incremental index is effectively a giant hashmap, a lot more work needs to be done before we can build v9 segments without conversion